### PR TITLE
[pyOpenMS] simple helper to update scores from a previously exported dataframe

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Use case is e.g., rescoring with ML methods.
e.g., user could write a simple script to export id date from idXML to df, run a DL method to calculate an updated score, feed it back into the OpenMS datastructures, write idXML out.
This could be plugged into e.g., larger workflows

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/pyOpenMS/tests/unittests/test000.py [hunk 1]
```diff
@@ -2296,10 +2297,16 @@ def test_peptide_identifications_to_df():
...
-    assert pyopenms.peptide_identifications_to_df(peps, decode_ontology=False).shape == (2,10)
+    assert pyopenms.peptide_identifications_to_df(peps).shape == (2,12)
+    assert pyopenms.peptide_identifications_to_df(peps, decode_ontology=False).shape == (2,12)
     assert pyopenms.peptide_identifications_to_df(peps)['protein_accession'][0] == 'sp|Accession1,sp|Accession2'
-    assert pyopenms.peptide_identifications_to_df(peps, export_unidentified=False).shape == (1,10)
+    assert pyopenms.peptide_identifications_to_df(peps, export_unidentified=False).shape == (1,12)
+
+    # update from dataframe
+    df = pyopenms.peptide_identifications_to_df(peps)
+    df["ScoreType"][0] = 10.0;
```
Comment (jpfeuffer) [inline src/pyOpenMS/tests/unittests/test000.py:2307]:
```suggestion
    df["ScoreType"][0] = 10.0
```

File: src/pyOpenMS/pyopenms/_dataframes.py [hunk 2]
```diff
@@ -516,4 +519,34 @@ def extract(pep):
                 ret.append(default_missing_values[type(val)])
         return tuple(ret)
 
-    return _pd.DataFrame(_np.fromiter((extract(pep) for pep in peps), dtype=dt, count=count))
+    return _pd.DataFrame(_np.fromiter((extract(pep, pep_idx) for pep_idx, pep in enumerate(peps)), dtype=dt, count=count))
+
+
+def update_scores_from_df(peps: List[_PeptideIdentification], df : _pd.DataFrame, main_score_name : str):
+    """Updates the scores in PeptideIdentification objects using a pandas dataframe.
+    Parameters:
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/_dataframes.py:527]: Did you check if this renders properly? We usually go for :param xyz: these days.

Do you need/want to specify the types twice?

Comment (timosachsenberg) [reply src/pyOpenMS/pyopenms/_dataframes.py:527]: no clue I just copied the style from the ones above.  I can look into it

Comment (jpfeuffer) [reply src/pyOpenMS/pyopenms/_dataframes.py]: I think the aliased classes are not documented correctly anyway: https://github.com/OpenMS/pyopenms-docs/issues/326

Comment (Arslan-Siraj) [reply src/pyOpenMS/pyopenms/_dataframes.py]: can look into it.

**Pull request merged**